stdlib/public/core/Slab.swift

```diff
@@ -0,0 +1,479 @@
+  @available(SwiftStdlib 6.1, *)
+  @_alwaysEmitIntoClient
+  @_transparent
+  public var indices: Range<Int> {
+    Range(_uncheckedBounds: (0, count))
+  }
+
+  /// Returns the position immediately after the given index.
+  ///
+  /// - Parameter i: A valid index of the collection. `i` must be less than
+  ///   `endIndex`.
+  /// - Returns: The index immediately after `i`.
+  @available(SwiftStdlib 6.1, *)
+  @_alwaysEmitIntoClient
+  @_transparent
+  public borrowing func index(after i: Int) -> Int {
+    i + 1
+  }
+
+  /// Returns the position immediately before the given index.
+  ///
+  /// - Parameter i: A valid index of the collection. `i` must be greater than
+  ///   `startIndex`.
+  /// - Returns: The index value immediately before `i`.
+  @available(SwiftStdlib 6.1, *)
+  @_alwaysEmitIntoClient
+  @_transparent
+  public borrowing func index(before i: Int) -> Int {
+    i - 1
+  }
+
+  /// Accesses the element at the specified position.
+  ///
+  /// The following example accesses an element of an array through its
+  /// subscript to print its value:
+  ///
+  ///     var streets = ["Adams", "Bryant", "Channing", "Douglas", "Evarts"]
+  ///     print(streets[1])
+  ///     // Prints "Bryant"
+  ///
+  /// You can subscript a collection with any valid index other than the
+  /// collection's end index. The end index refers to the position one past
+  /// the last element of a collection, so it doesn't correspond with an
+  /// element.
+  ///
+  /// - Parameter position: The position of the element to access. `position`
+  ///   must be a valid index of the collection that is not equal to the
+  ///   `endIndex` property.
+  ///
+  /// - Complexity: O(1)
+  @available(SwiftStdlib 6.1, *)
+  @_addressableSelf
+  @_alwaysEmitIntoClient
+  public subscript(_ i: Int) -> Element {
+    @_transparent
+    unsafeAddress {
+      _precondition(indices.contains(i), "Index out of bounds")
+
+      return address + i
+    }
+
+    @_transparent
+    unsafeMutableAddress {
+      _precondition(indices.contains(i), "Index out of bounds")
+
+      return mutableAddress + i
+    }
+  }
+}
+
+//===----------------------------------------------------------------------===//
+// Reduce and Swap
+//===----------------------------------------------------------------------===//
+
+@available(SwiftStdlib 6.1, *)
+extension Slab where Element: ~Copyable {
+  /// Exchanges the values at the specified indices of the vector.
+  ///
+  /// Both parameters must be valid indices of the vector and not
+  /// equal to `endIndex`. Passing the same index as both `i` and `j` has no
+  /// effect.
+  ///
+  /// - Parameters:
+  ///   - i: The index of the first value to swap.
+  ///   - j: The index of the second value to swap.
+  ///
+  /// - Complexity: O(1)
+  @available(SwiftStdlib 6.1, *)
+  @_alwaysEmitIntoClient
+  public mutating func swapAt(
+    _ i: Int,
+    _ j: Int
+  ) {
+    guard i != j, indices.contains(i), indices.contains(j) else {
+      return
```

Comment:
Shouldn't this trap when either is an out-of-bounds index?

Reply:
It can? Does not trapping and doing nothing with out of bounds indices cause an issue or lead to a potential issue?

Reply:
It's a silent failure; that seems as bad as trapping really. The `MutableCollection`'s implementation traps, by virtue of using the indexing subscripts under the hood. `UMBP`'s uses `debugPrecondition`, as if it deferred to the indexing subscripts.